packages/client/src/utils/userUtils.ts

```diff
@@ -60,3 +61,19 @@ export function useUserName() {
     return getUserName(userDetails)
   })
 }
+
+export function useUserScopes() {
+  return useSelector<IStoreState, Scope[]>((state) => {
+    const { tokenPayload } = state.profile
+    return tokenPayload ? tokenPayload.scope : []
+  })
+}
+
+export function userHasScope(userScopes: Scope[], allowedScopes: Scope[]) {
+  return allowedScopes.every((scope) => userScopes.includes(scope))
+}
+
+export function useUserHasScope(allowedScopes: Scope[]) {
+  const userScopes = useUserScopes()
+  return userHasScope(userScopes, allowedScopes)
+}
```

Comment:
https://github.com/opencrvs/opencrvs-core/blob/e0d02579c4fc6751338e620da368d763afdeeee5/packages/client/src/hooks/useAuthorization.ts
refactor this hook when handling scopes